test/tstDetailsRay.cpp

```diff
@@ -158,3 +157,74 @@ BOOST_AUTO_TEST_CASE(intersects_box)
   BOOST_TEST(!intersects(Ray{{1, 1.5, 1}, {0, -1, -2.1}}, unit_box));
   BOOST_TEST(!intersects(Ray{{1, 1.5, 1}, {-2.1, -1, 0}}, unit_box));
 }
+
+// NOTE until boost 1.70 need to cast both operands when comparing floating
+// points
+BOOST_AUTO_TEST_CASE(overlap_distance_sphere,
+                     *boost::unit_test::tolerance(
+                         boost::test_tools::fpc::percent_tolerance(0.0001f)))
```

Comment:
```suggestion
                     *boost::unit_test::tolerance(1e-6f))

```
From [here](https://www.boost.org/doc/libs/1_68_0/libs/test/doc/html/boost_test/testing_tools/extended_comparison/floating_point.html), these are equivalent. I also tested it locally.